test/integration/authorise_application_test.rb

```diff
@@ -6,22 +6,44 @@ class AuthoriseApplicationTest < ActionDispatch::IntegrationTest
     @user = create(:user)
   end
 
-  should "not confirm the authorisation for a signed-out user" do
+  should "not confirm the authorisation until the user signs in" do
     visit "/oauth/authorize?response_type=code&client_id=#{@app.uid}&redirect_uri=#{@app.redirect_uri}"
     assert_response_contains("You need to sign in")
-    signin(@user)
 
-    refute Doorkeeper::AccessGrant.find_by(resource_owner_id: @user.id)
```

Comment:
Did you mean to remove this assertion? Without it, the test isn't checking that the access grant is only created after signin so the test description isn't quite accurate.